Set content type when it is not explicitly set

qubicllj opened the pull request:
I use rrd4j to generate picture and convert it to bytes array, "application/octet-stream" will make browser to download the picture which is not what i want.

qubicllj committed:
Set content type when it is not explicitly set

buildhive commented:
[reyez » ninja #280](https://buildhive.cloudbees.com/job/reyez/job/ninja/280/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[reyez » ninja #281](https://buildhive.cloudbees.com/job/reyez/job/ninja/281/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

raphaelbauer commented:
Hey qubic,

sorry for my last post - my bad - I looked at the wrong file revision. Your change makes 100% sense. Thanks for that!

Two more things:
Could you please 
- add a tiny testcase to ResultHandlerTest.java to make sure byte[] does not get a forced octet stream
- add the bug you fixed to changelog.md (ninja-core/site/markdown/developer/)?

This will really help us to make sure the bug does not reappear :)

If you have problems extending the testcase let me know. I think I can do that too asap...

Thanks again for your contribution!

Best,

Raphael

qubicllj committed:
Add result handler test for rendering image from bytes

Fix for byte[] get a forced octet stream when content type is already
explicitly set

buildhive commented:
[reyez » ninja #282](https://buildhive.cloudbees.com/job/reyez/job/ninja/282/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[reyez » ninja #283](https://buildhive.cloudbees.com/job/reyez/job/ninja/283/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

raphaelbauer commented:
absolutely perfect! thanks :+1:
